### PR TITLE
Fixes #1044

### LINKED ISSUE #1044: Tips: Could also respond to tax
"7% tax on $35.99" seems like a reasonable query to me.

-- http://duck.co/ia/view/tips

### PR DESCRIPTION
**Pull request by sagarhani**
This would fix the issue #1044

### CODE REVIEW

**Comment by mintsoft** [comment]
@sagarhani You'll also need to update the guard regex on https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1076/files#diff-e3f1f44d059a17d1029e176d0ebf6c78R27

**Comment by mintsoft** [comment]
@sagarhani I'd probably add some tests in to show off the new functionality:https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/t/Tips.t

File: lib/DDG/Goodie/Tips.pm [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 use DDG::Goodie;
 with 'DDG::GoodieRole::NumberStyler';
 
 # Yes, 'of' is very generic, the gaurd should kick back false positives very quickly.
-triggers any => 'tip', 'tips', 'of';
+triggers any => 'tip', 'tips', 'of', 'tax';
 
 primary_example_queries '20% tip on $21.63';
 secondary_example_queries '20 percent tip for a $20 bill';
+tertiary_example_queries '20% tax on $21.63 bill';
```
Comment (mattr555) [inline lib/DDG/Goodie/Tips.pm]: @sagarhani We don't have a metadata field called `tertiary_example_queries`, but you can make a list like:

```perl
secondary_example_queries '20 percent tip for a $20 bill', '20% tax on $21.63 bill';
```

**Comment by moollaza** [comment]
Ping! @sagarhani are you still interested in finishing up this PR?

**Comment by sagarhani** [comment]
Sure. :+1:
I need some time. :)

**Comment by mintsoft** [comment]
@sagarhani Just checking in, are you still interested in finishing this off?

**Comment by sagarhani** [comment]
Looks like I'm not able to figure this out.
Anyone else who is interested could fix this.

**Comment by moollaza** [comment]
>Looks like I'm not able to figure this out.

@sagarhani care to explain where you're stuck? It seems like this should be a fairly easy fix?

File: lib/DDG/Goodie/Tips.pm [hunk 2]
```diff
@@ -23,7 +24,7 @@ zci is_cached   => 1;
 my $number_re = number_style_regex();
 
 handle query_lc => sub {
-    return unless (/^(?<p>$number_re)(?: ?%| percent) (?:(?<do_tip>tip (?:on|for|of))|of)(?: an?)? (?<sign>[\$\-]?)(?<num>$number_re)(?: bill)?$/);
+    return unless (/^(?<p>$number_re)(?: ?%| percent) (?:(?<do_tip>tip|tax (?:on|for|of))|of)(?: an?)? (?<sign>[\$\-]?)(?<num>$number_re)(?: bill)?$/);
```
Comment (moollaza) [inline lib/DDG/Goodie/Tips.pm]: You need to wrap `tip|tax` in `()`. Right now you've entirely separated the regular expression in to either matching `tip` or `tax (?:on|for|of)...`

I highly suggest you use debuggex.com to help figure out how your regex is working: https://www.debuggex.com/r/tnXtclj7uUDOk156

**Pull request closed without merging**